tools/tools.go

```diff
@@ -346,87 +344,31 @@ func EnvmanRun(envstorePth, workDirPth string, cmdArgs []string, timeout time.Du
 	args := []string{"--loglevel", logLevel, "--path", envstorePth, "run"}
 	args = append(args, cmdArgs...)
 
-	if !configs.IsSecretFiltering {
-		command := command.NewWithStandardOuts("envman", args...).SetStdin(os.Stdin).SetDir(workDirPth)
-
-		if timeout <= 0 {
-			exitCode, err := command.RunAndReturnExitCode()
-			return exitCode, errors.WithStack(err)
-		}
-
-		// create a new process group for our process and its child processes
-		cmd := command.GetCmd()
-		cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
-
-		if err := cmd.Start(); err != nil {
-			return -1, errors.WithStack(err)
-		}
-
-		// Setpgid: true creates a new process group for cmd and its subprocesses
-		// this way cmd will not belong to its parent process group,
-		// cmd will not be killed when you hit ^C in your terminal
-		// to fix this, we listen and handle Interrupt signal manually
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, os.Interrupt)
-		go func() {
-			<-c
-			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
-				log.Warnf("Failed to kill process, error: %+v", errors.WithStack(err))
-				os.Exit(130)
-			}
-		}()
-		defer func() {
-			signal.Stop(c)
-		}()
-		//
-
-		timer := time.AfterFunc(timeout, func() {
-			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
-				log.Warnf("Failed to kill process, error: %+v", errors.WithStack(err))
-				os.Exit(130)
-			}
-		})
-
-		err := cmd.Wait()
+	var outWriter io.Writer
+	var errWriter io.Writer
 
-		timer.Stop()
-
-		exitCode := 0
-		if err != nil {
-			exitCode = 1
-			if exiterr, ok := err.(*exec.ExitError); ok {
-				if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
-					exitCode = status.ExitStatus()
-				}
-			}
-			if err.Error() == "signal: killed" {
-				err = errors.New("timeout")
+	if !configs.IsSecretFiltering {
+		outWriter = os.Stdout
+		errWriter = os.Stderr
+	} else {
+		var secretValues []string
+		for _, secret := range secrets {
+			key, value, err := secret.GetKeyValuePair()
+			if err != nil || len(value) < 1 || key == configs.IsSecretFilteringKey {
```

Comment:
what is `key == configs.IsSecretFilteringKey`?

Reply:
discussed